subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml

```diff
@@ -0,0 +1,277 @@
+        <para>In opposition, any type that is used in the following list is irrelevant to the ABI, and therefore should be declared as <literal>implementation</literal> dependencies:
+        </para>
+        <itemizedlist>
+            <listitem>types exclusively used in method bodies</listitem>
+            <listitem>types exclusively used in private members</listitem>
+            <listitem>types exclusively found in internal classes (future versions of Gradle will let you declare which packages belong to the public API)</listitem>
+        </itemizedlist>
+        <para>In the following sample, we can make the difference between an API dependency and an implementation dependency:</para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="Imports found in a class">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="imports"/>
+        </sample>
+        <para>We can see that our class imports third party classes, but imports alone won't tell us if a dependency is an API or implementation dependency.
+            For this, we need to look at the methods. The following method used <literal>HttpClient</literal> as a parameter, so it belongs to the public API:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the public API">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="api-dependency"/>
+        </sample>
+        <para>On the other hand, the <literal>ExceptionUtils</literal> type, coming from the <literal>commons-lang</literal> library, is only used in a method
+            body, so it's an implementation dependency:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="A dependency type is used in the method bodies only">
+            <sourcefile file="src/main/java/org/gradle/HttpClientWrapper.java" snippet="implementation-dependency"/>
+        </sample>
+        <para>Therefore, we can deduce that <literal>commons-httpclient</literal> is an API dependency, whereas <literal>commons-lang</literal> is an implementation
+            dependency, which directly translates into the build file:
+        </para>
+        <sample id="useJavaLibraryPlugin" dir="java-library/quickstart" title="Declaring API and implementation dependencies">
+            <sourcefile file="build.gradle" snippet="dependencies"/>
+        </sample>
+        <para>It's in general safer to assume that a dependency is an implementation dependency only: leakage of implementation types to consumers would
+            then directly lead to a compile error, which would be solved either by removing the type from the public API, or promoting the dependency as an
+            API dependency instead.
+        </para>
+    </section>
+
+    <section id="sec:java_library_configurations_graph">
+        <title>The Java library plugin configurations</title>
```

Comment:
Upper case `Java Library` vs lower case `Java library`. I think we should consistently use UC.

`Java library` -> `Java Library`